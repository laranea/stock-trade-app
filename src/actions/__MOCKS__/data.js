export const MOCK_STOCKS = [
  {
    dividendYield: 0,
    fixedDividend: 0,
    geometricMean: 0,
    id: 1,
    lastDividend: 0,
    lastPrice: 0,
    parValue: 100,
    peRatio: 0,
    stockSymbol: 'TEA',
    type: 'common',
    volumeWeightedStockPrice: 0
  },
  {
    dividendYield: 0,
    fixedDividend: 0,
    geometricMean: 0,
    id: 2,
    lastDividend: 8,
    lastPrice: 0,
    parValue: 100,
    peRatio: 0,
    stockSymbol: 'POP',
    type: 'common',
    volumeWeightedStockPrice: 0
  }
];

export const MOCK_TRADES = [
  {
    id: 1,
    price: 10,
    stockSymbol: 'TEA',
    quantity: 100
  },
  {
    id: 1,
    price: 9,
    stockSymbol: 'TEA',
    quantity: 300
  }
];

export const MOCK_TRADE = {
  price: 10,
  stockSymbol: 'TEA',
  quantity: 100
};

export const MOCK_RESULT = [
  {
    dividendYield: 0,
    fixedDividend: 0,
    geometricMean: 9.486832980505138,
    id: 1,
    lastDividend: 0,
    lastPrice: 10,
    parValue: 100,
    peRatio: 0,
    stockSymbol: 'TEA',
    type: 'common',
    volumeWeightedStockPrice: 9.25
  },
  {
    dividendYield: 0,
    fixedDividend: 0,
    geometricMean: 0,
    id: 2,
    lastDividend: 8,
    lastPrice: 0,
    parValue: 100,
    peRatio: 0,
    stockSymbol: 'POP',
    type: 'common',
    volumeWeightedStockPrice: 0
  }
];
