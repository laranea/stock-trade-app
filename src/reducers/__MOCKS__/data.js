import { v4 as uuid } from 'uuid';

export const MOCK_STOCKS = [
  {
    dividendYield: 0,
    fixedDividend: 0,
    geometricMean: 0,
    id: uuid(),
    lastDividend: 0,
    lastPrice: 0,
    parValue: 100,
    peRatio: 0,
    stockSymbol: 'TEA',
    type: 'common',
    volumeWeightedStockPrice: 0
  },
  {
    dividendYield: 0,
    fixedDividend: 0,
    geometricMean: 0,
    id: uuid(),
    lastDividend: 8,
    lastPrice: 0,
    parValue: 100,
    peRatio: 0,
    stockSymbol: 'POP',
    type: 'common',
    volumeWeightedStockPrice: 0
  },
  {
    dividendYield: 0,
    fixedDividend: 0,
    geometricMean: 0,
    id: uuid(),
    lastDividend: 23,
    lastPrice: 0,
    parValue: 60,
    peRatio: 0,
    stockSymbol: 'ALE',
    type: 'common',
    volumeWeightedStockPrice: 0
  },
  {
    dividendYield: 0,
    fixedDividend: 2,
    geometricMean: 0,
    id: uuid(),
    lastDividend: 8,
    lastPrice: 0,
    parValue: 100,
    peRatio: 0,
    stockSymbol: 'GIN',
    type: 'preferred',
    volumeWeightedStockPrice: 0
  },
  {
    dividendYield: 0,
    fixedDividend: 0,
    geometricMean: 0,
    id: uuid(),
    lastDividend: 13,
    lastPrice: 0,
    parValue: 250,
    peRatio: 0,
    stockSymbol: 'JOE',
    type: 'common',
    volumeWeightedStockPrice: 0
  }
];

export const MOCK_STOCK_SYMBOLS = MOCK_STOCKS.map(stock => {
  return {
    id: stock.id,
    stockSymbol: stock.stockSymbol
  };
});
