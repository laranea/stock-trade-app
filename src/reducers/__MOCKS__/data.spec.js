import { MOCK_STOCKS, MOCK_STOCK_SYMBOLS } from './data';

jest.mock('uuid', () => {
  return {
    v4: jest.fn(() => 1)
  };
});

describe('MOCK_STOCKS', () => {
  it('should match snapshot', () => {
    expect(MOCK_STOCKS).toMatchSnapshot();
  });
});

describe('MOCK_STOCK_SYMBOLS', () => {
  it('should match snapshot', () => {
    expect(MOCK_STOCK_SYMBOLS).toMatchSnapshot();
  });
});
