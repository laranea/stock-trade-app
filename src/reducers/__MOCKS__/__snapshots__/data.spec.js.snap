// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOCK_STOCK_SYMBOLS should match snapshot 1`] = `
Array [
  Object {
    "id": 1,
    "stockSymbol": "TEA",
  },
  Object {
    "id": 1,
    "stockSymbol": "POP",
  },
  Object {
    "id": 1,
    "stockSymbol": "ALE",
  },
  Object {
    "id": 1,
    "stockSymbol": "GIN",
  },
  Object {
    "id": 1,
    "stockSymbol": "JOE",
  },
]
`;

exports[`MOCK_STOCKS should match snapshot 1`] = `
Array [
  Object {
    "dividendYield": 0,
    "fixedDividend": 0,
    "geometricMean": 0,
    "id": 1,
    "lastDividend": 0,
    "lastPrice": 0,
    "parValue": 100,
    "peRatio": 0,
    "stockSymbol": "TEA",
    "type": "common",
    "volumeWeightedStockPrice": 0,
  },
  Object {
    "dividendYield": 0,
    "fixedDividend": 0,
    "geometricMean": 0,
    "id": 1,
    "lastDividend": 8,
    "lastPrice": 0,
    "parValue": 100,
    "peRatio": 0,
    "stockSymbol": "POP",
    "type": "common",
    "volumeWeightedStockPrice": 0,
  },
  Object {
    "dividendYield": 0,
    "fixedDividend": 0,
    "geometricMean": 0,
    "id": 1,
    "lastDividend": 23,
    "lastPrice": 0,
    "parValue": 60,
    "peRatio": 0,
    "stockSymbol": "ALE",
    "type": "common",
    "volumeWeightedStockPrice": 0,
  },
  Object {
    "dividendYield": 0,
    "fixedDividend": 2,
    "geometricMean": 0,
    "id": 1,
    "lastDividend": 8,
    "lastPrice": 0,
    "parValue": 100,
    "peRatio": 0,
    "stockSymbol": "GIN",
    "type": "preferred",
    "volumeWeightedStockPrice": 0,
  },
  Object {
    "dividendYield": 0,
    "fixedDividend": 0,
    "geometricMean": 0,
    "id": 1,
    "lastDividend": 13,
    "lastPrice": 0,
    "parValue": 250,
    "peRatio": 0,
    "stockSymbol": "JOE",
    "type": "common",
    "volumeWeightedStockPrice": 0,
  },
]
`;
